crates/viewer/re_dataframe_ui/src/filters/boolean_filter.rs

```diff
@@ -0,0 +1,144 @@
+use arrow::datatypes::{DataType, Field};
+use datafusion::common::Column;
+use datafusion::logical_expr::{Expr, col, lit};
+use datafusion::prelude::{array_element, array_has, array_sort};
+use re_ui::UiExt as _;
+use re_ui::syntax_highlighting::SyntaxHighlightedBuilder;
+
+use super::{FilterError, FilterUiAction, Nullability};
+
+/// A filter for a boolean column.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BooleanFilter {
+    /// Filter for strictly non-nullable columns.
+    ///
+    /// This includes non-nullable primitive datatypes, as well as outer AND inner non-nullable nested
+    /// datatypes. In that case, the UI should not display the `null` option.
+    NonNullable(bool),
+
+    /// Filter for nullable columns.
+    ///
+    /// In this case, the UI should display a `null` option. A value of `None` means nulls should be
+    /// matched.
+    Nullable(Option<bool>),
```

Comment:
So does `Nullable(true)` shows nullable columns, or columns with nulls in it?

Reply:
For clarity, `FilterOperation` and related sub-struct such as `BooleanFilter` are only half of a filter. The full filter is `Filter`, with combines a column name with a filter operation.